packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -486,6 +492,10 @@ export class SearchPage extends Component {
               controller={this.redirectionTrigger}
             ></RedirectionTriggerFn>
           </Section>
+          <Section title="query-trigger">
+            <QueryTrigger></QueryTrigger>
+            <QueryTriggerFn controller={this.queryTrigger}></QueryTriggerFn>
+          </Section>
```

Comment:
It may be worth the time to create a branch that contains both these changes and https://github.com/coveo/ui-kit/pull/863 to test whether server-side rendering interacts well with this functionality.